### PR TITLE
Add ConfigMap resolver

### PR DESCRIPTION
**Pull request by matthchr**
Part of ongoing work related to #2317 and #2435.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/pkg/genruntime/arm_transformer.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ type ConvertToARMResolvedDetails struct {
 	// ResolvedSecrets is a set of secret references which have been resolved to the corresponding
 	// secret value.
 	ResolvedSecrets Resolved[SecretReference]
+
+	// ResolvedSecrets is a set of secret references which have been resolved to the corresponding
```
Comment (matthchr) [inline v2/pkg/genruntime/arm_transformer.go:26]: minor: fix

File: v2/internal/resolver/config_map_resolver.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			err := NewConfigMapNotFoundError(refNamespacedName, err)
+			return "", errors.WithStack(err)
+		}
+
+		return "", errors.Wrapf(err, "couldn't resolve config map reference %s", ref.String())
+	}
+
+	value, ok := configMap.Data[ref.Key] // TODO: Do we need to also check the binaryData field?
+	if !ok {
+		return "", NewSecretNotFoundError(refNamespacedName, errors.Errorf("ConfigMap %q does not contain key %q", refNamespacedName.String(), ref.Key))
```
Comment (super-harsh) [inline v2/internal/resolver/config_map_resolver.go:61]: `NewSecretNotFoundError` maybe we need one for configmap? or a generic one for both?

Comment (super-harsh) [reply v2/internal/resolver/config_map_resolver.go:61]: Oh I see you got `NewConfigMapNotFoundError` already.

**Review by super-harsh — APPROVED** [review]
LGTM. Just a minor change request.

File: v2/internal/resolver/config_map_resolver.go (hunk 2 above)
Comment (matthchr) [reply v2/internal/resolver/config_map_resolver.go:61]: Ooh good catch, this should be using the new error type.

**Pull request merged**